src/requirements.txt

```diff
@@ -1,3 +1,6 @@
 cumulus-message-adapter==2.0.3
 cumulus-message-adapter-python==2.2.0
+jsonpath-ng==1.4.2
+jsonschema==2.6.0
 mandible @ git+https://github.com/asfadmin/mandible@v0.4.0
+ply
```

Comment:
Needs to be pinned to a version as per our coding standards.

Reply:
@mattp0  asked me to remove the pin. So, should I pin it, or should I not?

Reply:
You two can debate that, I will add it back or leave it as it is.

Reply:
https://github.com/asfadmin/ia-standards/blob/main/standards.md#dependencies
> Dependencies must be pinned using one of the supported [version specifiers](https://packaging.python.org/en/latest/specifications/version-specifiers/#id4).

There is nothing to debate. I think it may be a good idea to have a read through the coding standards every once in a while to help reinforce what's in there because I keep seeing things in our PRs that just blatantly go against our standards.

Reply:
For the record, my PR did not blatantly go against our coding standards to begin with. I think it is good practice to look at previous conversations before telling me to go review the coding standards. I updated it when Matt made a suggestion that makes sense. If it is a requirement that isn't from our code, it does make more sense to not pin it. I will revert back to pinning it and if we can move this conversation to the coding standards repo if need be.

Reply:
> @mattp0 asked me to remove the pin. So, should I pin it, or should I not?

I think what Matt meant was to remove it from the requirements file entirely, not simply to unpin it. As I commented above though, the requirements file needs to list every requirement explicitly as the Makefile installs them with `--no-deps` to keep the code zip size smaller by ignoring unused packages.

Reply:
I don't think that is the case. This was the comment:
`I don not see ply being used.. If its a dependency of anther package we shouldnt pin it imo .`

Reply:
> For the record, my PR did not blatantly go against our coding standards to begin with.

I know. However, in the state it was merged it goes against the standards. I just don't see how that could happen if everyone was clear about what the standards are, as in theory anyway, PRs that don't meet the standards shouldn't be merged.

Reply:
While I generally agree with this statement, I do think there are cases we didn't consider while writing the coding standards. In my opinion, we need to be agile in these kinds of situations. This is one of the cases I think Matt had a good point and pinning a dependency that isn't ours doesn't seem like the correct call. I did create a PR to revert the change.

Reply:
I would be happy to talk more about dependency pinning with the team as I think it's actually quite an interesting subject. 

There is actually a great reason why you might pin a dependency that is not explicitly used in your project: reproducible builds. In fact, we do this in TEA, however, there we use an automated tool called `pip-compile` to take a list of just our direct dependencies and produce a flattened list of the entire dependency tree with the exact versions pinned to ensure that every time we do a build, we get exactly the same code pulled in.

Reply:
FWIW: My comment was more a double check that it should be there. Based on the conversation we had @reweeden , I thought we should rely on the packages you are calling to handle dependency pinning rather than explicitly pinning something. Was maybe a mis understanding on my part, but if we arent pinning the whole dep tree with pip-compile it seemed like it could lead to issues of the package that depends on it updated with this left behind.

Reply:
> I thought we should rely on the packages you are calling to handle dependency pinning rather than explicitly pinning something. 

It all depends on what context you are working in. Is your code a library that will be used by others or is it a standalone application? If you're writing a library, you want your requirements list to be as flexible as possible so you should only include your direct dependencies but if you're writing an application (as we are here) you want to pin your requirements down as much as possible so that every time you build a particular commit, you get the same result.



Was maybe a mis understanding on my part, but if we arent pinning the whole dep tree with pip-compile it seemed like it could lead to issues of the package that uses ply is updated with this left behind.



> if we arent pinning the whole dep tree with pip-compile it seemed like it could lead to issues of the package that uses ply is updated with this left behind.

Correct. Since we're doing it manually here to trim out the unused dependencies that `pip-compile` would bring in, you would need to take care that all of the versions of the packages continue to work together. We may be able to actually revisit this because when I was putting together `cnm-to-granules` I created issues on each of the github repos of packages that had unused dependencies in their requirements, and I think most or maybe all of those have been fixed by now in later versions of the packages.